nexus-sled-agent-shared/src/inventory.rs

```diff
@@ -331,6 +333,38 @@ impl ZoneImageResolverInventory {
             mupdate_override: MupdateOverrideInventory::new_fake(),
         }
     }
+
+    /// Returns a displayer for this inventory.
+    pub fn display(&self) -> ZoneImageResolverInventoryDisplay<'_> {
+        ZoneImageResolverInventoryDisplay { inner: self }
+    }
+}
+
+/// Displayer for a [`ZoneImageResolverInventory`]
+pub struct ZoneImageResolverInventoryDisplay<'a> {
+    inner: &'a ZoneImageResolverInventory,
+}
+
+impl fmt::Display for ZoneImageResolverInventoryDisplay<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let ZoneImageResolverInventory { zone_manifest, mupdate_override } =
+            self.inner;
+
+        writeln!(f, "zone manifest:")?;
+        let mut indented = IndentWriter::new("    ", f);
+        // Use write! rather than writeln! because zone_manifest.display()
+        // always produces a newline at the end.
+        write!(indented, "{}", zone_manifest.display())?;
+        let f = indented.into_inner();
+
+        writeln!(f, "mupdate override:")?;
+        let mut indented = IndentWriter::new("    ", f);
```

Comment:
Could get a ref to the inner writer to avoid needing to recreate `indented`?

```suggestion
        writeln!(indented.get_ref(), "mupdate override:")?;
```

Maybe that's actually less clear though. Your call

Reply:
`get_ref` returns a shared reference, but we need a mutable ref here unfortunately.